### ISSUE TITLE
CTL feedback should be more explicit about which particular test failed

### ISSUE THREAD

**Issue by nmco** [opengeospatial/ets-wmts10#40]
Consider the following tests execution report, this is what we get when we click on a test details from the UI:

![image](https://user-images.githubusercontent.com/1590542/33274164-350aaf04-d387-11e7-9d29-2f091db18a38.png)

```
Test wmts:Server.KVP.GET.GetTile.Missing type Mandatory default result Passed (s0022/d27092e418_1/d7936e6014_1/d7936e6912_2)

Assertion: 
	1.  the request entity is a valid KVP GET GetTile request with the mandatory request parameter with a value of GetTile;
	2.  the request entity is a valid KVP GET GetTile request with the mandatory service parameter with a value of WMTS;
	3.  the request entity is a valid KVP GET GetTile request with the mandatory Version parameter with a value of 1.0.0;
	4.  the request entity is a valid KVP GET GetTile request with the mandatory Layer parameter with a layer identifier value equal to a <Layer><ows:Identifier> value from the Service Metadata document for layers with no sample dimensions, or with sample dimensions with a default dimension from the Service Metadata document; except when tested as missing;
	5.  the request entity is a valid KVP GET GetTile request with the mandatory Style parameter with a style identifier value equal to a <Layer><Style><owsIdentifier> value for the specified Layer from the Service Metadata document, except when tested as missing;
	6.  the request entity is a valid KVP GET GetTile request with the mandatory Format parameter parameter with a MIME type format value equal to a <Layer><Format> value for the specified Layer from the Service Metadata document, except when tested as missing;
	7.  the request entity is a valid KVP GET GetTile request with the mandatory TileMatrixSet parameterwith a TileMatrixSet identifier value equal to a <Layer><TileMatrixSetLink><TileMatrixSet> value for the specified Layer from the Service Metadata document, except when tested as missing;
	8.  the request entity is a valid KVP GET GetTile request with the mandatory TileMatrix parameter with a TileMatrix identifier value equal to a <TileMatrixSet><TileMatrix><owsIdentifier> value for the specified TileMatrixSet from the Service Metadata document, except when tested as missing;
	9.  the request entity is a valid KVP GET GetTile request with the mandatory TileRow parameter with a TileRow index value from zero to the value of <MatrixHeight> minus 1 for the specified TileMatrix from the Service Metadata document, except when tested as missing;
	10.  the request entity is a valid KVP GET GetTile request with the mandatory TileCol parameter with a TileCol index value from zero to the value of <MatrixWidth> minus 1 for the specified TileMatrix from the Service Metadata document, except when tested as missing;
	11.  the request entity for a tile from a Layer with TileMatrixSetLimits with TileMatrixLimits that references the identifier of the TileMatrix for the tile specifies a TileRow index value within the TileMatrixLimits MinTileRow and MaxTileRow values, except when tested as missing.
	12.  the request entity for a tile from a Layer with TileMatrixSetLimits with TileMatrixLimits that references the identifier of the TileMatrix for the tile specifies a TileCol index value within the TileMatrixLimits MinTileCol and MaxTileCol values, except when tested as missing.
	13.  the response entity is a schema-valid ows:Exception document.
	14.  the response entity contains an exceptionCode attribute value of &quot;MissingParameterValue&quot;.
	15.  the response entity contains a locator attribute value of the name of the missing parameter.
	16.  the HTTP status code value is set to 400, corresponding to the message &quot;Bad Request&quot;.

Context:
   Label: 
   Value: wmts:main

Message d7936e16453_1:
   Issuing KVP GET GetTile requests to http://localhost:8085/geoserver/gwc/service/wmts for the following:

Message d7936e16683_1:
   layer MISSING  style polygon format image/png TileMatrixSet EPSG:4326 has limits true TileMatrix EPSG:4326:10 row 300 col 478

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_1:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&style=polygon&format=image%2Fpng&TileMatrixSet=EPSG:4326&TileMatrix=EPSG:4326:10&TileRow=300&TileCol=478
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">446</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="LAYER">
            <ExceptionText>Missing LAYER parameter</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [8] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators layer

Message d5728e976_1:
   exception.validation.elapsed [4] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1048_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator LAYER valid for specified locator layer

Message d5728e1112_1:
   Exception report contains required valid exception code(s) and locator(s)

Message d5728e1121_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16683_2:
   layer sf:states style MISSING  format image/png TileMatrixSet EPSG:4326 has limits true TileMatrix EPSG:4326:10 row 300 col 478

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_2:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&layer=sf:states&format=image%2Fpng&TileMatrixSet=EPSG:4326&TileMatrix=EPSG:4326:10&TileRow=300&TileCol=478
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">470</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="request">
            <ExceptionText>Mandatory STYLE query parameter not provided.</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [7] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators style

Message d5728e976_1:
   exception.validation.elapsed [3] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1056_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator request not valid for specified locator style

Message d5728e1088_1:
   Exception report does not contain required valid exception code(s) and locator(s)

Message d5728e1100_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16683_3:
   layer sf:states style polygon format MISSING  TileMatrixSet EPSG:4326 has limits true TileMatrix EPSG:4326:10 row 300 col 478

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_3:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&layer=sf:states&style=polygon&TileMatrixSet=EPSG:4326&TileMatrix=EPSG:4326:10&TileRow=300&TileCol=478
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">466</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="FORMAT">
            <ExceptionText>Unable to determine requested FORMAT, null</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [6] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators format

Message d5728e976_1:
   exception.validation.elapsed [3] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1048_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator FORMAT valid for specified locator format

Message d5728e1112_1:
   Exception report contains required valid exception code(s) and locator(s)

Message d5728e1121_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16683_4:
   layer sf:states style polygon format image/png TileMatrixSet MISSING  has limits true TileMatrix EPSG:4326:10 row 300 col 478

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_4:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&layer=sf:states&style=polygon&format=image%2Fpng&TileMatrix=EPSG:4326:10&TileRow=300&TileCol=478
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">457</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="TILEMATRIXSET">
            <ExceptionText>No TILEMATRIXSET specified</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [9] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators TileMatrixSet

Message d5728e976_1:
   exception.validation.elapsed [3] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1048_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator TILEMATRIXSET valid for specified locator TileMatrixSet

Message d5728e1112_1:
   Exception report contains required valid exception code(s) and locator(s)

Message d5728e1121_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16683_5:
   layer sf:states style polygon format image/png TileMatrixSet EPSG:4326 has limits true TileMatrix MISSING  row 300 col 478

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_5:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&layer=sf:states&style=polygon&format=image%2Fpng&TileMatrixSet=EPSG:4326&TileRow=300&TileCol=478
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">451</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="TILEMATRIX">
            <ExceptionText>No TILEMATRIX specified</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [9] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators TileMatrix

Message d5728e976_1:
   exception.validation.elapsed [2] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1048_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator TILEMATRIX valid for specified locator TileMatrix

Message d5728e1112_1:
   Exception report contains required valid exception code(s) and locator(s)

Message d5728e1121_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16683_6:
   layer sf:states style polygon format image/png TileMatrixSet EPSG:4326 has limits true TileMatrix EPSG:4326:10 row MISSING  col 478

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_6:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&layer=sf:states&style=polygon&format=image%2Fpng&TileMatrixSet=EPSG:4326&TileMatrix=EPSG:4326:10&TileCol=478
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">445</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="TILEROW">
            <ExceptionText>No TILEROW specified</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [8] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators TileRow

Message d5728e976_1:
   exception.validation.elapsed [2] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1048_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator TILEROW valid for specified locator TileRow

Message d5728e1112_1:
   Exception report contains required valid exception code(s) and locator(s)

Message d5728e1121_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16683_7:
   layer sf:states style polygon format image/png TileMatrixSet EPSG:4326 has limits true TileMatrix EPSG:4326:10 row 300 col MISSING 

Request 21f4fdc6/63a9ad1/7229644b/d7936e16687_7:
   Method: get
   URL: http://localhost:8085/geoserver/gwc/service/wmts?service=WMTS&request=GetTile&version=1.0.0&layer=sf:states&style=polygon&format=image%2Fpng&TileMatrixSet=EPSG:4326&TileMatrix=EPSG:4326:10&TileRow=300
   Response from parser parsers:HTTPParser:
      <response>
   <status code="400" protocol="HTTP/1.1">Bad Request</status>
   <headers>
      <header name="Date">Sat, 25 Nov 2017 23:15:57 GMT</header>
      <header name="X-Frame-Options">SAMEORIGIN</header>
      <header name="Content-Type">text/xml</header>
      <header name="Content-Disposition">inline; filename=geoserver-dispatch.text</header>
      <header name="Content-Length">445</header>
      <header name="Server">Jetty(9.2.13.v20150730)</header>
   </headers>
   <content>
      <ExceptionReport xmlns="http://www.opengis.net/ows/1.1"
                       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                       version="1.1.0"
                       xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://geowebcache.org/schema/ows/1.1.0/owsExceptionReport.xsd">
         <Exception exceptionCode="MissingParameterValue" locator="TILECOL">
            <ExceptionText>No TILECOL specified</ExceptionText>
         </Exception>
      </ExceptionReport>
   </content>
   <!--Response received in [13] milliseconds--></response>

Message d5728e910_1:
   Validating ows:ExceptionReport against its XML Schema xsd/ogc/ows/1.1.0/owsExceptionReport.xsd for codes MissingParameterValue locators TileCol

Message d5728e976_1:
   exception.validation.elapsed [3] milliseconds

Message d5728e981_1:
   Exception report is XML Schema valid, continuing...

Message d5728e1048_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator TILECOL valid for specified locator TileCol

Message d5728e1112_1:
   Exception report contains required valid exception code(s) and locator(s)

Message d5728e1121_1:
   Required HTTP status code 400 was returned by server.

Message d7936e16856_1:
   Test failed.  Server.KVP.GET.GetTile.Missing

Result: Failed
```

One of the tests failed because there is a wrong ``locator`` that was used in the exception report. The spot this error we have to go carefully thought all the log above and step on this lines:

```
Message d5728e1056_1:
   The returned exception report contained expected exceptionCode MissingParameterValue with locator request not valid for specified locator style

Message d5728e1088_1:
   Exception report does not contain required valid exception code(s) and locator(s)
```

This could be improved by adding a tag like ``FAILURE`` to the beginning of the message, the _nirvana_ would be to only see the relevant error messages.

**Comment by keshav-nangare** [comment]
Added the [FAILURE] tag before the message. 


Fixed with the commit: 84d6bd792d3d59e58016b4a237f46896ebf74814 [PR: #44]

**Comment by dstenger** [comment]
@keshav-nangare Can you please resolve the merge conflicts of https://github.com/opengeospatial/ets-wmts10/pull/44#issuecomment-387435253?
